### PR TITLE
WIP: Vulkanize the sampler

### PR DESCRIPTION
**Pull request by msiglreith**
Fixes https://github.com/gfx-rs/gfx/issues/1404
PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [ ] tested examples with the following backends:

### CODE REVIEW

File: src/hal/src/image.rs [hunk 1]
```diff
@@ -208,29 +208,29 @@ impl AaMode {
...
-///
-/// These names are somewhat poor, in that "bilinear" is really just doing
-/// linear filtering on each axis, and it is only bilinear in the case of 2D
-/// textures. Similarly for trilinear, it is really Quadralinear(?) for 3D
-/// textures. Alas, these names are simple, and match certain intuitions
-/// ingrained by many years of public use of inaccurate terminology.
+/// but also cost.
 #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum FilterMethod {
```
Comment (kvark) [inline src/hal/src/image.rs]: please note that there is a `BlitFilter` type that can now be removed as a duplicate

Comment (kvark) [reply src/hal/src/image.rs]: also, we might want to shorten it to just `Filter` as well as map the values directly to Vulkan counterparts for FFI

Comment (msiglreith) [reply src/hal/src/image.rs]: thanks, done

**Comment by msiglreith** [comment]
Vulkan, DX12 and GL should be complete now. Could someone take care of the Metal backend?

**Comment by kvark** [comment]
@msiglreith awesome! if no one else, I'll take care of Metal tonight.

**Comment by kvark** [comment]
Finished all the sampler states in Metal and ported to the new `SamplerInfo`. Interestingly, I don't see setting LOD bias on the sampler descriptor in the API reference, but it seems to pass with no complaints.
Also verified the building/reftesting/examples work on Metal. Time to sail!
bors r+

**Pull request merged**